### PR TITLE
update wasmtime to 0.24.0

### PR DESCRIPTION
**Pull request by ordian**
To partially unblock build on darwin-aarch64 (aka Apple M1 Silicon).

Note that the [changelog](https://github.com/bytecodealliance/wasmtime/blob/main/RELEASES.md) includes interesting changes, such as 
* https://github.com/bytecodealliance/wasmtime/pull/2617

which we might want to use. However, it is out of scope of this PR.

polkadot companion: https://github.com/paritytech/polkadot/pull/2625

### CODE REVIEW

**Comment by bkchr** [comment]
This requires a burn in. Full sync of Kusama and Polkadot

File: frame/support/test/tests/derive_no_bound_ui/eq.stderr [hunk 1]
```diff
@@ -4,4 +4,9 @@ error[E0277]: can't compare `Foo<T>` with `Foo<T>`
 6   | struct Foo<T: Config> {
     |        ^^^ no implementation for `Foo<T> == Foo<T>`
     |
+   ::: $RUST/core/src/cmp.rs
```
Comment (bkchr) [inline frame/support/test/tests/derive_no_bound_ui/eq.stderr:7]: Please revert

Comment (ordian) [reply frame/support/test/tests/derive_no_bound_ui/eq.stderr:7]: hmm, I wonder why it fails locally, UI tests are using stable, right? So it should be the same 1.50 version.

**Comment by gnunicorn** [comment]
anyone working on this?

**Comment by ordian** [comment]
@gnunicorn I can update the companion once the burnin is complete

**Review by rphmeier — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Pull request merged**